app/src/main/java/com/android/voyageur/model/place/GooglePlacesRepository.kt

```diff
@@ -74,22 +90,46 @@ class GooglePlacesRepository(private val placesClient: PlacesClient) : PlacesRep
         .addOnFailureListener { exception -> onFailure(exception) }
   }
 
+  /**
+   * Fetches detailed information for a list of place IDs.
+   *
+   * @param placeIds The list of place IDs to fetch details for.
+   * @param onSuccess Callback invoked with the list of detailed places.
+   * @param onFailure Callback invoked with an exception if the fetch fails.
+   */
   private fun fetchPlaceDetails(
       placeIds: List<String>,
-      onSuccess: (List<Place>) -> Unit,
+      onSuccess: (List<CustomPlace>) -> Unit,
       onFailure: (Exception) -> Unit
   ) {
-    val places = mutableListOf<Place>()
+    val places = mutableListOf<CustomPlace>()
     var failedRequests = 0
 
     // Define which fields to retrieve for each place
-
     placeIds.forEach { placeId ->
       val request = FetchPlaceRequest.builder(placeId, placeFields).build()
       placesClient
           .fetchPlace(request)
           .addOnSuccessListener { response ->
-            places.add(response.place)
+            val place = response.place
+            val bitmaps = mutableListOf<ImageBitmap>()
+
+            val photosMetadata =
+                (place.photoMetadatas?.map { it } ?: emptyList<PhotoMetadata>()).take(3)
+            for (photoMetadata in photosMetadata) {
+              val photoRequest = FetchPhotoRequest.builder(photoMetadata).build()
+              placesClient
+                  .fetchPhoto(photoRequest)
+                  .addOnSuccessListener { fetchPhotoResponse ->
+                    val bitmap = fetchPhotoResponse.bitmap.asImageBitmap()
+                    bitmaps.add(bitmap)
+                  }
+                  .addOnFailureListener { exception ->
+                    Log.e("PlacesRepository", "Failed to fetch photo", exception)
```

Comment:
It might be helpful to include the placeId in the error log message for .addOnFailureListener. This would make it easier to identify which place's photo failed to load, making debugging easier.

Reply:
Done